rust/agama-utils/src/question.rs

```diff
@@ -52,9 +52,72 @@
 //! # });
 //! ```
 pub mod service;
+use std::time::Duration;
+
 pub use service::Service;
 
 pub mod message;
 
 pub mod start;
 pub use start::start;
+use tokio::time::sleep;
+
+use crate::{
+    actor::Handler,
+    api::question::{Answer, QuestionSpec},
+    question,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum AskError {
+    #[error("Question not found")]
+    QuestionNotFound,
+    #[error(transparent)]
+    Service(#[from] question::service::Error),
+    #[error(transparent)]
+    Actor(#[from] crate::actor::Error),
+}
+
+/// Asks a question and waits until it is answered.
+///
+/// This is a helper function for internal Rust services that have a direct handler to the
+/// `question::Service` and need to ask a question and wait for the answer in a
+/// synchronous-like manner within an `async` context. It simplifies the process by
+/// abstracting away the need to listen for events or poll the API, which would
+/// typically be done by an external client (like a web UI).
+///
+/// The function sends the question to the `question::Service` and then polls for an
+/// answer. Once the question is answered (either by a user, a pre-configured rule,
+/// or a default policy), the function returns the answer and deletes the question
+/// from the service to clean up.
+///
+/// # Arguments
+/// * `handler` - A handler to the `question::Service`.
+/// * `question` - The `QuestionSpec` defining the question to be asked.
+///
+/// # Errors
+/// This function can return the following errors:
+/// * `AskError::QuestionNotFound`: If the question is deleted by another process before it can be answered.
```

Comment:
```suggestion
/// * [AskError::QuestionNotFound]: If the question is deleted by another process before it can be answered.
```

If you use this notation, it will link to `AskError::QuestionNotFound`. It applies to all the references on this module.